packages/analytics/src/Consent.ts

```diff
@@ -0,0 +1,79 @@
+import { CONSENT_KEYS } from './utils';
+import merge from 'lodash/merge';
+import pick from 'lodash/pick';
+import type { ConsentData } from './types/analytics.types';
+import type StorageWrapper from './utils/StorageWrapper';
+
+const defaultConsentData: ConsentData = {
+  marketing: false,
+  preferences: false,
+  statistics: false,
+};
+
+/**
+ * Stores the consent values in a new instance.
+ */
+class Consent {
+  storage: StorageWrapper;
+  /**
+   * Constructs a new consent instance with the passed in storage wrapper instance.
+   *
+   * @param storage - The storage wrapper instance where data will be stored.
+   */
+  constructor(storage: StorageWrapper) {
+    // NOTE: For now, we will only check if the storage reference is set to something,
+    //      as we are already validating the storage on the analytics class, so it is not
+    //      required to do it right now. If we expose this class to be used in other
+    //      scenarios, then we will need to revisit this validation code.
+    if (!storage) {
+      throw new Error(
+        'Invalid storage instance provided to Consent constructor',
+      );
+    }
+
+    this.storage = storage;
+  }
+
+  /**
+   * Returns the stored consent object. If not defined, returns the defaults.
+   *
+   * @returns Promise that will resolve with the consent data.
+   */
+  async get(): Promise<ConsentData> {
+    const getItem = (await this.storage.getItem(
+      'consent',
+    )) as Partial<ConsentData>;
+
+    return {
+      marketing: getItem?.marketing || defaultConsentData.marketing,
```

Comment:
Here we cannot return the default values if nothing returns from the storage. The functionality is:
If there is no consent given, this method must return `null`. If the user interacted with the consent (allowing these categories or not), we return what was previously stored.
The return `null` case allows tenants to understand if the user did interact with the consent already or not (e.g.: "Accept all cookies" banner, otherwise we wouldn't have a way to understand if the user didn't give permissions at all (that would lead to have all categories with "false" value, which is the default as well).

Reply:
Yes, my bad, haven't checked against the previous implementation. Reverted, thanks!